### PR TITLE
Add HTTP and S3 health checks, add deleteObject method to the S3 wrapper

### PR DESCRIPTION
**Pull request by szeber**

### CODE REVIEW

File: src/HealthCheck/HttpHealthCheck.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
+<?php
+declare(strict_types=1);
```
Comment (emulgeator) [inline src/HealthCheck/HttpHealthCheck.php]: Was this intentional?

File: src/HealthCheck/HttpHealthCheck.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        $url          = $this->getUrl($configOptions);
+        $result       = $this->sendRequest($configOptions, $url);
+        $responseBody = $result->getResponseBody();
+
+        $this->checkStatus($configOptions, $result, $url);
+        $this->checkJson($configOptions, $responseBody);
+        $this->checkBody($configOptions, $responseBody);
+    }
+
+    protected function checkArrayContents(array $expected, array $array): bool
```
Comment (emulgeator) [inline src/HealthCheck/HttpHealthCheck.php]: This method is not clear to me (can you find a better name?), but it looks somethign general what should go into another class.
Also some unit tests would be awesome for this

File: src/HealthCheck/HttpHealthCheck.php (hunk 1 above)
Comment (szeber) [reply src/HealthCheck/HttpHealthCheck.php]: We did this for all the health checks. In theory we are not really violating anything with it, it's new stuff, that does require a higher version of php... We should probably migrate to a minimum version of 7.1 and switch versions to 0.3.

File: src/HealthCheck/HttpHealthCheck.php (hunk 2 above)
Comment (szeber) [reply src/HealthCheck/HttpHealthCheck.php]: Some tests in general would be good, so we should probably test the whole project properly.

**Pull request merged**